community/kernel/src/main/java/org/neo4j/kernel/impl/store/record/ConstraintRule.java

```diff
@@ -43,9 +43,26 @@ public static ConstraintRule constraintRule(
         return new ConstraintRule( id, descriptor, ownedIndexRule );
     }
 
+    public static ConstraintRule constraintRule(
+            long id, ConstraintDescriptor descriptor, String name )
+    {
+        return new ConstraintRule( id, descriptor, null, name );
```

Comment:
I was thinking to generate the default name here, but whatever. Having it in SchemaRule brings the functionality together, and that's always nice.